src/test/java/org/reaktivity/nukleus/tcp/internal/streams/ServerPartialWriteIT.java

```diff
@@ -0,0 +1,172 @@
+    })
+    @BMUnitConfig(loadDirectory="src/test/resources", debug=false, verbose=false)
+    @BMScript(value="PartialWriteIT.btm")
+    public void shouldFinishWriteWhenSocketIsWritableAgain() throws Exception
+    {
+        PartialWriteBytemanHelper.addWriteResult(5);
+        shouldReceiveServerSentData("server data");
+    }
+
+    @Test
+    @Specification({
+        "${route}/input/new/controller",
+        "${streams}/server.sent.data/server/target"
+    })
+    @BMUnitConfig(loadDirectory="src/test/resources", debug=false, verbose=false)
+    @BMScript(value="PartialWriteIT.btm")
+    public void shouldHandleMultiplePartialWrites() throws Exception
+    {
+        PartialWriteBytemanHelper.addWriteResult(2);
+        PartialWriteBytemanHelper.addWriteResult(3);
+        PartialWriteBytemanHelper.addWriteResult(1);
+        shouldReceiveServerSentData("server data");
+    }
+
+    @Test
+    @Specification({
+        "${route}/input/new/controller",
+        "${streams}/server.sent.data.multiple.frames/server/target"
+    })
+    @BMUnitConfig(loadDirectory="src/test/resources", debug=false, verbose=false)
+    @BMScript(value="PartialWriteIT.btm")
+    public void shouldWriteWhenMoreDataArrivesWhileAwaitingSocketWritable() throws Exception
+    {
+        PartialWriteBytemanHelper.addWriteResult(5);
+        PartialWriteBytemanHelper.addWriteResult(8);
+        PartialWriteBytemanHelper.addWriteResult(5);
+        PartialWriteBytemanHelper.addWriteResult(5);
+        shouldReceiveServerSentData("server data 1server data 2");
+
+        // Verify we forced the desired condition: check handleWrite got called
+        // AFTER processData was called for the second frame
+        List<String> callers = PartialWriteBytemanHelper.callers();
+        String error = "Test failed to force desired condition, caller sequence was: " + callers;
+        assertTrue(error, callers.lastIndexOf("handleWrite") >  callers.lastIndexOf("processData"));
+    }
+
+    // TODO: shouldResetStreamsExceedingPartialWriteStreamsLimit
+
+    // TODO: Write exceeds window (requires new spec tests, should probably go in ServiceIT and ClientIT)
+
+    private void shouldReceiveServerSentData(String expectedData) throws Exception
+    {
+        k3po.start();
+        k3po.awaitBarrier("ROUTED_INPUT");
+
+        try (Socket socket = new Socket("127.0.0.1", 0x1f90))
+        {
+            final InputStream in = socket.getInputStream();
+
+            byte[] buf = new byte[expectedData.length() + 10];
+            int offset = 0;
+
+            int read = 0;
+            do
+            {
+                read = in.read(buf, offset, buf.length - offset);
+                if (read == -1)
+                {
+                    break;
+                }
+                offset += read;
+            } while (offset < expectedData.length());
+            assertEquals(expectedData, new String(buf, 0, offset, UTF_8));
+        }
+
+        k3po.finish();
+    }
```

Comment:
I don't see the point, please explain. Does this concern one particular test method?